wip/integrationguide.md

```diff
@@ -0,0 +1,110 @@
+# Integration Guide:  Porting Mimi Models from v0.4.0 to v0.5.0
+
+## Overview
+
+The release of Mimi v0.5.0 is a breaking release, necessitating the adaptation of existing models' syntax and structure in order for those models to run on this new version.  This guide provides an overview of the steps required to get most models using the v0.4.0 API working with v0.5.0.  It is **not** a comprehensive review of all changes and new functionalities, but a guide to the minimum stpes required to port old models between versions.  For complete information on the new version and it's functionalities, see the full documentation.
+
+This guide is organized into six main sections, each descripting an independent set of changes that can be undertaken in any order desired.  For clarity, these sections echo the organization of the `userguide`.
+
+1) Defining components
+2) Constructing a model
+3) Running the model
+4) Accessing results
+5) Plotting
+6) Advanced topics
+
+## Defining Components
+
+The `run_timestep` function is now contained by the `@defcomp` macro, and takes the parameters `p, v, d, t`, referring to Parameters, Variables, and Dimensions of the component you defined.  The fourth argument is a Timestep, which represents which timestep the model is at each time the function gets called.  Similarly, the optional `init` function is also contained by `@defcomp`, and takes the parameters `p, v, d`.  Thus, as described in the user guide, definiing a single component is now done as follows:
+
+```julia
+@defcomp component1 begin
+
+    # First define the state this component will hold
+    savingsrate = Parameter()
+
+    # Second, define the (optional) init function for the component
+    function init(p, v, d)
+    end
+
+    # Third, define the run_timestep function for the component
+    function run_timestep(p, v, d, t)
+    end
+
+end
+```
+
+## Constructing a Model
+
+In an effort to standardize the function naming protocol within Mimi, and to streamline it with the standards of Julia, several function names have been changed.  The table below lists a **subset** of these changes, focused on the exported API functions most commonly used in model construction.  
+
+| Old Syntax                | New Syntax                |
+| ------------------------  |:-------------------------:|
+|`connectparameter`         |`connect_parameter`        |
+|`setleftoverparameters`    |`set_leftover_params!`     |
+|`setparameter`             |`set_parameter!`           |
+|`adddimension`             |`add_dimension`            |
+|`setindex`                 |`set_dimension!`           |  
```

Comment:
Should it be `add_dimension!`?

Reply:
@rjplevin no, this is actually currently `add_dimension` ... perhaps just because it adds one but does not set it like `set_dimension!`?  If `add_dimension!` follows convention better I can change that.

Reply:
I was thinking it should have the "!" for consistency, since it modifies the model definition.

Reply:
I'd tend to agree, I can make a small PR to change this